Feature/1.x/146 url aliases

ekes opened the pull request:
Test with an update to HEAD of group_domain 
and on a content type with url alias field enabled, which is all localgov types I found other than the new page one for which I'm making another PR.

Should allow alias to be set for the domain on which the content is being edited.
Will allow multiple aliases to be set should content be in multiple domains (current set up actually already allows two, one for the control site and one for the group domain in the case you are editing content that is in a domain group but doing so on the control domain).

ekes committed:
Enable domain path for #146

ekes committed:
Merge branch '1.x' into feature/1.x/146-url-aliases

ekes committed:
CS: Remove unused use statement.

finnlewis commented:
On first test, I get an error when trying to add a url alias to a new directory channel.

I did a fresh install.
Created microsites 1
Created admin 1
Assigned admin1 to microsite 1
Logged into microsite 1
Create a directory facet type
Create a directory channel
Give it an alias of /contact
I get the following error: 

```
The website encountered an unexpected error. Please try again later.

Drupal\Core\Entity\EntityMalformedException: The "node" entity cannot have a URI as it does not have an ID in Drupal\Core\Entity\EntityBase->toUrl() (line 161 of core/lib/Drupal/Core/Entity/EntityBase.php).
Drupal\domain_path\DomainPathHelper->submitEntityForm(Array, Object)
call_user_func_array(Array, Array) (Line: 114)
Drupal\Core\Form\FormSubmitter->executeSubmitHandlers(Array, Object) (Line: 52)
Drupal\Core\Form\FormSubmitter->doSubmitForm(Array, Object) (Line: 592)
Drupal\Core\Form\FormBuilder->processForm('node_localgov_directory_form', Array, Object) (Line: 320)
Drupal\Core\Form\FormBuilder->buildForm(Object, Object) (Line: 48)
Drupal\Core\Entity\EntityFormBuilder->getForm(Object, 'default', Array) (Line: 368)
Drupal\group\Entity\Controller\GroupContentController->createForm(Object, 'group_node:localgov_directory')
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 564)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 158)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 80)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 58)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 48)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 51)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 709)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

I'm out for a bit but can test more later this afternoon.

ekes commented:
Confirmed: localgov_page works; localgov_directories_channel errors.

ekes commented:
> Confirmed: localgov_page works; localgov_directories_channel errors.

It's because it uses the two step form. Which in the first case we can fix by stopping using the two step form; but I'll see if I can't make a patch for domain_path to stop it trying to create an alias before the content is actually saved.

ekes commented:
https://www.drupal.org/project/domain_path/issues/3293894

I guess, except for rewriting the module, we make some patch for ourselves that overrides the form alter in the case of what looks like a group two-step form and sets a message somewhere that it won't work with domain_path.

ekes committed:
Add patch to notify of failure if using groupcontent wizard.

Domain Path doesn't support multi-step forms. Issue https://www.drupal.org/project/domain_path/issues/3293894
Solution outside of rewriting the module is to set a notification for
our specific case of group multistep wizard forms https://github.com/localgovdrupal/localgov_microsites/pull/175#issuecomment-1172879196

ekes committed:
Add patch 995261d for #146

stephen-cox commented on config/install/user.role.authenticated.yml:
```diff
@@ -29,6 +30,7 @@ permissions:
   - 'edit any document media'
   - 'edit any image media'
   - 'edit any remote_video media'
+  - 'edit domain path entity'
```
I'm proposing we remove the authenticated user role config #168 

These permissions should be added to [microsites_trusted_editor](https://github.com/localgovdrupal/localgov_microsites_group/blob/1.x/config/install/user.role.microsites_trusted_editor.yml) and [microsites_controller](https://github.com/localgovdrupal/localgov_microsites_group/blob/1.x/config/install/user.role.microsites_controller.yml) as required.

ekes replied:
Which way round we do this?

ekes replied:
Just checking the permission. This is required to set a path on an entity. Is this only microsites_trusted_editor, or is it anyone who can post content?

stephen-cox commented on modules/localgov_microsites_media/localgov_microsites_media.module:
```diff
@@ -6,7 +6,6 @@
  */
 
 use Drupal\Core\Session\AccountInterface;
-use Drupal\domain_group\DomainGroupHelper;
```
This should have been removed in the 1.x branch.

stephen-cox reviewed (changes requested):
Functionally this is working fine. Just a minor issue about where to put the permissions for this.

ekes committed:
Merge branch '1.x' into feature/1.x/146-url-aliases

ekes committed:
Merge remote-tracking branch 'origin/1.x' into feature/1.x/146-url-aliases

ekes committed:
Don't add permissions to authenticated.

https://github.com/localgovdrupal/localgov_microsites/pull/175#pullrequestreview-1032762388
and
https://github.com/localgovdrupal/localgov_microsites/pull/175

ekes commented:
@stephen-cox assume we can get this is now https://github.com/localgovdrupal/localgov_microsites_group/commit/454c8487bf008a7529f42a582e82c356873c0b73 is in

stephen-cox reviewed (approved):
Great - let's get it in
